common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsTokenRequest.java

```diff
@@ -37,8 +38,18 @@ public class MicrosoftStsTokenRequest extends MicrosoftTokenRequest {
     /**
      * Additional JWT claims which would be used to create JWT based token requests
      */
-    private transient Map<String, String> mJwtClaims;
+
+    private transient AbstractMicrosoftStsTokenResponseHandler mTokenResponseHandler;
```

Comment:
I don't think a component like this in the code, should be part of a token request object (even if transient).

Reply:
Perhaps this should a member variable of `OAuth2Strategy` class?

Reply:
OAuth2Strategy's are classified at provider level (MSSTS (v2), AAD V1 etc), so we should be able to use strategy itself of different requests. Expected response handling really depends on the request rather than the OAuth2Strategy itself. So, I prefer to not keep response handling it at class level.

Assuming we want to keep MicrosoftStsTokenRequest as data class, then I am adding a new object MicrosoftStsTokenRequestContext which also holds request context like response handler for client tracking and processing